cmd/subnetcmd/upgradecmd/apply.go

```diff
@@ -204,29 +200,151 @@ func applyLocalNetworkUpgrade(subnetName string, sc models.Sidecar) error {
 		ux.Logger.PrintToUser("The next upgrade will go into effect %s", time.Unix(nextUpgrade, 0).Local().Format(constants.TimeParseLayout))
 		ux.PrintTableEndpoints(clusterInfo)
 
-		// it seems all went well this far, now we try to write/update the lock file
-		// if this fails, we probably don't want to cause an error to the user?
-		// so we are silently failing, just write a log entry
-		wrapper := params.UpgradeConfig{
-			PrecompileUpgrades: precmpUpgrades,
+		return writeLockFile(precmpUpgrades, subnetName)
+	}
+
+	return errors.New("unexpected network size of zero nodes")
+}
+
+// applyPublicNetworkUpgrade applies an upgrade file to a locally running validator
+// for public networks (fuji, main)
+// the validation of the upgrade file has many things to consider:
+// * No upgrade file for <public net> can be found - do we copy the existing file in the prev stage?
+// (for fuji: take the local, for main, take the fuji?)?
+// * If not, we exit, but then force the user to create a fuji file? Can be quite annoying!
+// * Do we validate that the fuji file is the same as local before applying? Or we just take whatever is there?
+// For main, that it's the same as fuji and/or local or take whatever is there?
+// * What if the local deployment has applied different stages of upgrades,
+// but they were for development only and fuji/main is going to be different (start from scratch)?
+// * What if someone isn't even doing local, just fuji and main...(or even just main...we may want to discourage that though...)
+// * User probably would never use the exact same file for local as for Fuji, because you’d probably want to change the timestamps
+//
+// For public networks we therefore limit ourselves to just "apply" the upgrades
+// This also means we are *ignoring* the lock file here!
+func applyPublicNetworkUpgrade(subnetName, networkKey string, sc *models.Sidecar) error {
+	if print {
+		blockchainIDstr := "<your-blockchain-id>"
+		if sc.Networks != nil &&
+			sc.Networks[networkKey] != (models.NetworkData{}) &&
```

Comment:
is this needed if blockchain id is not empty?

Reply:
No but it is if it _is_ empty